.pre-commit-config.yaml

```diff
@@ -0,0 +1,47 @@
+fail_fast: false
+default_language_version:
+    python: python3
+default_stages:
+-   commit
+-   push
+minimum_pre_commit_version: 2.10.1
+repos:
+-   repo: https://github.com/psf/black
+    rev: 22.3.0
+    hooks:
+    -   id: black
+        additional_dependencies: [toml]
+-   repo: https://github.com/timothycrosley/isort
+    rev: 5.7.0
+    hooks:
+    -   id: isort
+        additional_dependencies: [toml]
+        args: [--order-by-type]
+-   repo: https://github.com/myint/autoflake
+    rev: v1.4
+    hooks:
+    -   id: autoflake
+        exclude: .*notebooks.*
+        args: [--in-place, --remove-all-unused-imports, --remove-unused-variable, --ignore-init-module-imports]
+-   repo: https://gitlab.com/pycqa/flake8
+    rev: 3.8.4
+    hooks:
+    -   id: flake8
+        args: 
+        - --select=E9,F63,F7,F82
+        - --max-line-length=127
+        exclude: .*notebooks.*
+        additional_dependencies: [flake8-tidy-imports, flake8-docstrings, flake8-rst-docstrings, flake8-comprehensions, flake8-bugbear, flake8-logging-format, flake8-blind-except, flake8-builtins, flake8-pytest-style, flake8-mock, flake8-string-format]
```

Comment:
Do we need all the dependencies?

Reply:
Afaik `additional_dependencies` needs to still be specified in the pre-commit config, since they are plugins.
If you wanna prune them, take a look at https://github.com/scverse/cookiecutter-scverse/blob/main/%7B%7Bcookiecutter.project_name%7D%7D/.pre-commit-config.yaml#L34-L40